### PR TITLE
Add tech spec template to discussion

### PR DESCRIPTION
**Pull request by Caleb-Cohen**
# Description

Should add a tech spec template, but idk. 

## Testing

Describe what a reviewer should do to test your code, including instructions for manually testing as well as relevant automated test suites to run.

## Type of change

Please select everything applicable.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change requires an update to testin

# Checklist:

- [ ] Changes have new/updated automated tests
- [ ] Changes have been manually tested
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by timmyichen — APPROVED** [review]
🎉

File: .github/DISCUSSION_TEMPLATE/technical-specification.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+name: Technical Specification
+about: High level architecture for new systems/features
+title: 'Tech spec: '
+labels: tech spec
```
Comment (timmyichen) [inline .github/DISCUSSION_TEMPLATE/technical-specification.md:5]: I think this needs to be `technical-specification-review` instead

**Review by timmyichen — CHANGES_REQUESTED** [review]
Oops - actually I think we need to change the label here

**Comment by Caleb-Cohen** [comment]
Hello everyone, good morning.

File: .github/DISCUSSION_TEMPLATE/technical-specification.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+A technical specification exists to guide implementation and to clear any ambiguity around what needs to be done in order to ship a project. The level of detail should be enough so that people won't be wondering _how_ something will be done from a high level, but it shouldn't be so much that the tech spec is basically doing the entire project.
+
+When writing technical documents, err on the side of brevity. Say just enough to get your point across.
+
+When writing tech specs, err on the side of decreasing scope instead of increasing scope if faced with the two choices.
+
+Not every section requires a response. Just leave `N/A` under a section if it's not applicable.
+
+**Delete the text above before submitting**
```
Comment (devonzara) [inline .github/DISCUSSION_TEMPLATE/technical-specification.md:20]: Perhaps this should read "**Delete this and the text above before submitting.**"...

File: .github/DISCUSSION_TEMPLATE/technical-specification.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Summary:
+
+Provide a short (1-2 sentences) overview of the project, including its purpose and goals.
+
+## Scope:
+
+Define the scope of the project, including what is within scope and what is out of scope.
+
+## Data model:
```
Comment (devonzara) [inline .github/DISCUSSION_TEMPLATE/technical-specification.md:30]: How nit-picky are we wanting to be with reviews? 🤔

Capitalize "Model"?

**Review by timmyichen — APPROVED** [review]
TY for making the change! Approval from me but I'll defer to the others re the other comments

**Review by MatthewBozin — APPROVED** [review]

**Comment by Caleb-Cohen** [comment]
> TY for making the change! Approval from me but I'll defer to the others re the other comments

I'm actually not sure this works. I tested it out at Together and didn't seem to do anything.

**Comment by timmyichen** [comment]
> I'm actually not sure this works. I tested it out at Together and didn't seem to do anything.

Just looked into it, looks like the template is different - YAML, not markdown, but allows for greater customization: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema

**Comment by MainlyColors** [comment]
> > I'm actually not sure this works. I tested it out at Together and didn't seem to do anything.
> 
> Just looked into it, looks like the template is different - YAML, not markdown, but allows for greater customization: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema

So does this need to be converted to YAML because the linked article mentions Markdown is still possible?

> You can use both Markdown and YAML issue templates in your repository. If you want to convert a Markdown issue template to a YAML issue form template - Linked Article

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms#converting-a-markdown-issue-template-to-a-yaml-issue-form-template

**Comment by timmyichen** [comment]
> So does this need to be converted to YAML because the linked article mentions Markdown is still possible?

Caleb got something working in a test repo, he'll have updates here soon I think!

**Comment by Caleb-Cohen** [comment]
> > So does this need to be converted to YAML because the linked article mentions Markdown is still possible?
> 
> Caleb got something working in a test repo, he'll have updates here soon I think!

done

**Comment by timmyichen** [comment]
> done, looks pretty dank

Share a screenshot! We as reviewers have no way of verifying what it looks like 🙃

**Comment by Caleb-Cohen** [comment]
> > done, looks pretty dank
> 
> Share a screenshot! We as reviewers have no way of verifying what it looks like 🙃

Dank is a objective and testable fact, why require a photo? It's dank regardless. 

Screenshot 2023-03-13 at 1 45 06 PM

**Review by devonzara — APPROVED** [review]
Looks good!

**Review by timmyichen — APPROVED** [review]
dank levels are off the charts

**Review by code2244 — APPROVED** [review]
so dank!

**Comment by Caleb-Cohen** [comment]
LET THERE BE DISCUSSION TEMPLATES

**Pull request merged**